pkg/ec2provider/ec2provider.go

```diff
@@ -277,3 +285,41 @@ func (p *ec2ProviderImpl) getPrivateDnsAndPublishToCache(instanceIdList []string
 		p.unsetRequestInFlightForInstanceId(id)
 	}
 }
+
+func applySTSRequestHeaders(stsClient *sts.STS, roleARN, sourceARN string) *sts.STS {
+	logrus.Infof("Using AWS assumed role %v", roleARN)
+	sourceAcct, err := getSourceAccount(roleARN)
+	if err != nil {
+		logrus.Errorf("failed to parse source account from role ARN %v: %v", roleARN, err)
+		return stsClient
```

Comment:
Good callout, found we have check in the server lifecycle earlier https://github.com/kubernetes-sigs/aws-iam-authenticator/blob/master/pkg/server/server.go#L195-L200. I would prefer still keep the check here for robustness regardless in case new caller invokes EC2 provider function, but this check will be no-op currently